// petpage/src/app/layout.tsx
import type { Metadata } from "next";
import { Montserrat_Alternates } from "next/font/google";
import "./globals.css";
import { AosInit } from "./_components/aos-init";



// const geistSans = Geist({
//   variable: "--font-geist-sans",
//   subsets: ["latin"],
// });

const montserratAlternates = Montserrat_Alternates({
  subsets: ['latin'], // Quais subsets de caracteres você quer incluir (ex: latim)
  weight: ['400', '700'], // Quais pesos da fonte você quer usar (ex: regular e negrito)
  // Ou se você quiser todos os pesos disponíveis:
  // weight: 'variable',
});

// const geistMono = Geist_Mono({
//   variable: "--font-geist-mono",
//   subsets: ["latin"],
// });

export const metadata: Metadata = {
  title: "Create Next App",
  description: "Generated by create next app",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body
        className={`${montserratAlternates.className}  antialiased`}
      >
        {children}
        <AosInit/>
      </body>
    </html>
  );
}
