// petCorner/src/firebase.ts
// firebase.ts
import { initializeApp, type FirebaseApp } from "firebase/app";
import { getAuth, type Auth, GoogleAuthProvider, OAuthProvider } from "firebase/auth";
import { getFirestore, type Firestore } from "firebase/firestore";

let appPromise: Promise<FirebaseApp> | null = null;

function loadFirebaseConfig() {
  return fetch("/api/env")
    .then((res) => {
      if (!res.ok) throw new Error("Erro ao carregar variáveis de ambiente do Firebase");
      return res.json();
    })
    .then((env) => ({
      apiKey: env.FIREBASE_API_KEY,
      authDomain: env.FIREBASE_AUTH_DOMAIN,
      projectId: env.FIREBASE_PROJECT_ID,
      storageBucket: env.FIREBASE_STORAGE_BUCKET,
      messagingSenderId: env.FIREBASE_MESSAGING_SENDER_ID,
      appId: env.FIREBASE_APP_ID,
      measurementId: env.FIREBASE_MEASUREMENT_ID,
    }));
}

export function getFirebaseApp(): Promise<FirebaseApp> {
  if (!appPromise) {
    appPromise = loadFirebaseConfig().then((config) => initializeApp(config));
  }
  return appPromise;
}

// 🔥 Retorna Promise<Auth>
export async function getFirebaseAuth(): Promise<Auth> {
  const app = await getFirebaseApp();
  return getAuth(app);
}

// 🔥 Retorna Promise<Firestore>
export async function getFirestoreDB(): Promise<Firestore> {
  const app = await getFirebaseApp();
  return getFirestore(app);
}

export const googleProvider = new GoogleAuthProvider();
export const microsoftProvider = new OAuthProvider("microsoft.com");
